demonstrations_v2/tutorial_active_volume/demo.py

```diff
@@ -0,0 +1,712 @@
+additional qubits in the execution of the CNOT. This may come as a surprise, given that
+CNOT is a simple two-qubit Clifford gate. However, note that the active volume computer
+architecture operates with :doc:`lattice surgery <demos/tutorial_lattice_surgery>`, and
+executing a CNOT with lattice surgery also requires two additional logical qubits as intermediate
+bridge space. Thus, the oriented ZX diagram simply reproduces the true cost for a CNOT in this
+type of architecture.
+
+Note that we could also connect the input state :math:`|b\rangle` to the D port of the left X spider.
+This is because we could have inserted the additional blue spider on the output leg of the existing
+one, rather than between the inserted orange and the existing blue spiders.
+For this, the orientation of the spiders could be kept as-is.
+
+Oriented ZX diagram of the CNOT ladder
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Having figured out the oriented ZX diagram for a single CNOT gate, we turn to the ZX diagram
+of the CNOT ladder. As you may guess, compared to the unoriented diagram we arrived at above, we
+will again need to insert more nodes.
+One way to do this would be to simply copy the oriented diagram for the single CNOT three times
+in the vertical (time) direction, and to route outputs of one copy into the inputs of the next.
+This would lead to an arrangement with "depth" three.
+However, here we want to keep the full circuit in a single layer. With a similar reasoning as for
+the single CNOT, we find that we need to insert six nodes overall, arriving at the oriented ZX diagram:
+
+.. figure:: _static/demonstration_assets/active_volume/example_oriented_zx_complete.png
+    :align: center
+    :width: 90%
+    :target: javascript:void(0)
+
+In order to connect the “individual” CNOT networks side-by-side, we have moved some of the input
+qubit legs to what was previously an “internal” node. This is allowed as there is no preference
+for where to place logical inputs and outputs in a network. Note how some of the qubits appear
+to be routed laterally through the network, hinting at the parallelization of what were previously
+sequential gates. As for the single CNOT, this is not a unique solution.
+We are almost done with the parallelization of the CNOT ladder; one last step will be needed to
+transform the oriented ZX diagram into a logical network, which can then be executed on an
+active volume computer.
+
+Oriented ZX diagram to logical network
+--------------------------------------
+
+The last step required to arrive at logical networks is comparatively small; the major effort
+is already done with the orientation of the ZX diagram.
+
+We introduce **logical blocks**, which are just the nodes from an oriented ZX diagram,
+drawn as hexagons to mark the six ports more clearly, with a modification of rule (4.) above:
+
+    4'. Edges (internal legs) must connect to **the same port** at both logical blocks.
```

Comment:
It's still unclear to me what a logical block is 😬 Isnt rule 4' describing a logical block? What is meant that edges must connect to the same port at **both** blocks?

Like, is a logical block one of the hexagonal spiders? or a collection of them?

Reply:
> Like, is a logical block one of the hexagonal spiders? or a collection of them?

One logical block is really just one ZX spider from an oriented ZX diagram to which we apply the small modification from rule 4. to rule 4'. :) I'll rephrase to make the 1-to-1 correspondence explicit.

> What is meant that edges must connect to the same port at both blocks?

In 4., edges must connect to the same or the opposite ports of the two blocks they connect. In 4'., opposite ports are no longer good enough, the edge must connect the same port of both blocks. Would this be clearer ?

```suggestion
We introduce **logical blocks**, which are just the nodes from an oriented ZX diagram,
drawn as hexagons to mark the six ports more clearly, with a modification of rule (4.) above:

    4'. Edges (internal legs) must attach to **the same ports** at the two logical blocks they connect, e.g. N to N or W to W.
```

Reply:
Ah! So a logical block is always just a single spider?
Then perhaps simply:
> 4'. Edges (internal legs) to other logical blocks must attach to **the same ports**.

I just found it confusing that we are talking about a single logical block and then say something about "both blocks"